packages/xstate-react/src/useActor.ts

```diff
@@ -55,9 +23,8 @@ export function useActor(
     [actorRef]
   );
 
-  const boundGetSnapshot = useCallback(() => getSnapshot(actorRef), [
-    actorRef,
-    getSnapshot
+  const boundGetSnapshot = useCallback(() => actorRef.getSnapshot(), [
```

Comment:
Would it be better to pre-bind `.getSnapshot()`?

Reply:
If we start assuming that `actorRef` has been created by XState then we could consider that. There was no such assumption up until now though. In theory, the only thing that we require here is for the interface to be satisfied and that doesn't guarantee us if the `this` is being used in a method or not.

I would be fine with requiring "XState actors" here though - people using it with other objects of the same interface is rather a far-fetched scenario.